backend/src/main/java/heartbeat/controller/report/GenerateReportController.java

```diff
@@ -65,15 +68,79 @@ public InputStreamResource exportCSV(@PathVariable String dataType, @PathVariabl
 		return result;
 	}
 
-	@GetMapping("/{reportId}")
+	@GetMapping("/reports/{reportId}")
 	public ResponseEntity<ReportResponse> generateReport(@PathVariable String reportId) {
 		boolean generateReportIsOver = generateReporterService.checkGenerateReportIsDone(reportId);
+		ReportResponse reportResponse = generateReporterService.getComposedReportResponse(reportId,
+				generateReportIsOver);
 		if (generateReportIsOver) {
-			ReportResponse reportResponse = generateReporterService.getReportFromHandler(reportId);
 			log.info("Successfully generate Report, report id: {}, reports: {}", reportId, reportResponse);
+			// todo: calculate the time to generate csv file
+			generateReporterService.generateCSVForMetric(reportResponse, reportId);
 			return ResponseEntity.status(HttpStatus.CREATED).body(reportResponse);
 		}
-		return ResponseEntity.status(HttpStatus.NO_CONTENT).build();
+		return ResponseEntity.status(HttpStatus.OK).body(reportResponse);
+	}
+
+	@PostMapping("/board-reports")
```

Comment:
Wrong pattern for the restful design!
POST `reports/{reportType}`
Refactor the rest RESTful API

Reply:
done